tests/keras/metrics_test.py

```diff
@@ -173,9 +176,36 @@ def __call__(self, y_true, y_pred):
     def ref_true_pos(y_true, y_pred):
         return np.sum(np.logical_and(y_pred > 0.5, y_true == 1))
 
+    def ref_recall(y_true, y_pred):
+        return ref_true_pos(y_true, y_pred) / float(np.sum(y_true == 1))
+
+    def ref_precision(y_true, y_pred):
+        return ref_true_pos(y_true, y_pred) / float(np.sum(y_pred > .5))
+
     # Test correctness (e.g. updates should have been run)
-    np.testing.assert_allclose(outs[2], ref_true_pos(y, preds), atol=1e-5)
+    np.testing.assert_allclose(outs[2], ref_recall(y, preds), atol=1e-5)
+    np.testing.assert_allclose(outs[3], ref_precision(y, preds), atol=1e-5)
+    np.testing.assert_allclose(outs[4], ref_true_pos(y, preds), atol=1e-5)
+
```

Comment:
Look at the unit tests. The reference values are not computed with tensors and they match the tensor values (computed in batches).